### PR TITLE
Unknown type bugfix

### LINKED ISSUE #1316: Unable to load Mathesar when a column contains an unrecognized type
## Reproduce

1. Execute this SQL:

    ```sql
    DROP TABLE IF EXISTS "type_test";
    CREATE TABLE "type_test" (
        "id" serial NOT NULL,
        "array_of_ints" integer[]
    );
    INSERT INTO "type_test" ("array_of_ints") VALUES
    ('{ 1, 2, 3 }');
    ```

1. Restart Docker.

1. Try to load the base page.

1. Observe the following error:

    ```
    TypeError at /mathesar_tables/1/
    
    __init__() missing 1 required positional argument: 'item_type'

    ...
    ```

    image

## Notes

- I've tried this with a few different types, `integer[]`, `circle`, `jsonb`. I get the same behavior every time.
- After dropping the offending table, I'm able to load Mathesar again.

### PR DESCRIPTION
**Pull request by mathemancer**
Fixes #1316 

This adds a couple lines to smoothly ignore exceptions that may occur when a type isn't recognized by SQLAlchemy.

**Technical details**

Because we rely on the SQLAlchemy class representing a DB type for creating strings to represent the type in the API, problems can occur when no such class exists. This PR modifies this behavior to simply return `null` when we don't understand a type.

## Checklist
 
- [X] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [X] My pull request targets the `master` branch of the repository
- [X] My commit messages follow [best practices][best_practices].
- [X] My code follows the established code style of the repository.
- [X] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [X] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by seancolsen — APPROVED** [review]
This code works for me, given the steps I took to reproduce the problem. Thanks for the quick fix, @mathemancer 

I'll leave it to the backend team to decide if this PR needs further review before merging.

**Review by kgodey — APPROVED** [review]

**Pull request merged**
